src/main/flight/rpm_filter.c

```diff
@@ -70,46 +70,48 @@ void rpmFilterInit(const rpmFilterConfig_t *config, const timeUs_t looptimeUs)
 {
     motorIndex = 0;
     harmonicIndex = 0;
-    rpmFilter.numHarmonics = 0; // disable RPM Filtering
+    rpmFilter.activeHarmonicsCount = 0; // disable RPM Filtering
 
     // if bidirectional DShot is not available
     if (!useDshotTelemetry) {
         return;
     }
+    
+    for (int n = 0; n < RPM_FILTER_HARMONICS_MAX; n++) {
+        rpmFilter.weights[n] = constrainf(config->rpm_filter_weights[n] / 100.0f, 0.0f, 1.0f);
+        if (rpmFilter.weights[n] > 0.0f) {
+            rpmFilter.activeHarmonicsCount++;
+        }
+    }
 
     // if RPM Filtering is configured to be off
-    if (!config->rpm_filter_harmonics) {
+    if (!isRpmFilterEnabled()) {
         return;
     }
 
     // if we get to this point, enable and init RPM filtering
-    rpmFilter.numHarmonics = config->rpm_filter_harmonics;
     rpmFilter.minHz = config->rpm_filter_min_hz;
     rpmFilter.maxHz = 0.48f * 1e6f / looptimeUs; // don't go quite to nyquist to avoid oscillations
     rpmFilter.fadeRangeHz = config->rpm_filter_fade_range_hz;
     rpmFilter.q = config->rpm_filter_q / 100.0f;
     rpmFilter.looptimeUs = looptimeUs;
 
-    for (int n = 0; n < RPM_FILTER_HARMONICS_MAX; n++) {
-        rpmFilter.weights[n] = constrainf(config->rpm_filter_weights[n] / 100.0f, 0.0f, 1.0f);
-    }
-
     for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
         for (int motor = 0; motor < getMotorCount(); motor++) {
-            for (int i = 0; i < rpmFilter.numHarmonics; i++) {
-                biquadFilterInit(&rpmFilter.notch[axis][motor][i], rpmFilter.minHz * i, rpmFilter.looptimeUs, rpmFilter.q, FILTER_NOTCH, 0.0f);
+            for (int i = 0; i < RPM_FILTER_HARMONICS_MAX; i++) {
```

Comment:
It is possible to use only `activeHarmonicsCount` entries in rpmFilter.notch (skipping disabled harmonics), but it will slightly complicate code with small effect.
One advantage is that it would allow larger  rpmFilter.weights, with limit on nonzero entries (but is that useful?)